### DISCUSSION TITLE
How to parese mysql query slow log?

### DISCUSSION THREAD

**Discussion by spihiker** [vectordotdev/vector#20342] [General]
1.slow log 
# Time: 2024-04-19T08:14:30.240692Z
# User@Host: root[root] @ localhost []  Id: 695248
# Query_time: 5.000470  Lock_time: 0.000000 Rows_sent: 1  Rows_examined: 0
use uni_mydb;
SET timestamp=1713514470;
select sleep(5);

2.I want to convert the above text into json as follows。Can mysql slow query log be parsed? Please give an example. Thanks.

{“time“：“2024-04-19 16:40:30”,
 “user”:“root”,
 "host":"localhost",
 "query_time","5.000",
 "lock_time":"0.000",
 "rows_sent":1，
 "rows_examined”:0，
"sql_text”:"select sleep(5);"}

**Comment by jszwedko** [comment]
`parse_grok` or `parse_regex` is probably your best bet.

**Reply by spihiker** [reply]
Please give an example。Thanks.
